### ISSUE TITLE
[Bug] custom FPS does not work on servers without Server Boost or in DM without Nitro

### ISSUE THREAD

**Issue by fdkchx** [philhk/Vencord#21]
### Discord Account

kosmoxapka

### What happens when the bug or crash occurs?

custom FPS does not work on servers without Server Boost or in DM without Nitro

### What is the expected behaviour?

custom FPS will work again
for example, the CustomScreenShare plugin allows you to do this without any problems

### How do you recreate this bug or crash?

opened betterscreenshare settings, disabled simple, set framerate to 75.

### Errors

```
Replace this text with your crash-log.
```

### Request Agreement

- [X] I am using Discord Stable or tried on Stable and this bug happens there as well
- [X] I have read the requirements for opening an issue above

**Comment by Loukious** [comment]
Works fine for me. You just have to open the menu and press apply again.

**Comment by BlueberryWolf** [comment]
> Works fine for me. You just have to open the menu and press apply again.

I was able to replicate it. it only happens _specifically_ in servers with no boosts.

**Comment by BlueberryWolf** [comment]
> > Works fine for me. You just have to open the menu and press apply again.
> 
> I was able to replicate it. it only happens _specifically_ in servers with no boosts for me.

nevermind. it doesn't work for me at all in any server.

**Comment by Loukious** [comment]
> > > Works fine for me. You just have to open the menu and press apply again.
> > 
> > 
> > I was able to replicate it. it only happens _specifically_ in servers with no boosts for me.
> 
> nevermind. it doesn't work for me at all in any server.

Did you open the menu and press "apply" again AFTER you started streaming?
![image](https://github.com/philhk/Vencord/assets/24696530/b46212c7-a131-4da8-9695-285649c15e7f)

**Comment by BlueberryWolf** [comment]
yep, then discord immediately crashes for me. it only started occurring to me after I restarted my PC recently. Discord's console throws an error, "Invalid Framerate: 144"

**Comment by BlueberryWolf** [comment]
![image](https://github.com/philhk/Vencord/assets/19942815/1f4174ca-8a75-4719-ac7f-b5af3dba8449)
![image](https://github.com/philhk/Vencord/assets/19942815/3ec6211a-547b-4b5c-a551-1a19935aa1f4)

**Comment by BlueberryWolf** [comment]
![image](https://github.com/philhk/Vencord/assets/19942815/b078ff7d-5472-4826-ae57-070ee4185136)
here are my settings

**Comment by BlueberryWolf** [comment]
I've tried reinstalling stable discord, and also using the development branch. I'll try canary in a bit

**Comment by BlueberryWolf** [comment]
![image](https://github.com/philhk/Vencord/assets/19942815/1ec8a49c-609e-43b6-9eb6-52f62c55914e)

**Comment by BlueberryWolf** [comment]
I made a patch to fix it, it no longer crashes. I'll make a pull request in a second here.

All you have to do is return e on the default case.

**Comment by BlueberryWolf** [comment]
holy fuck the issue is bigger than I thought. if you even *view* a stream that is at a non-standard framerate, it crashes everyone else's discord

**Comment by Loukious** [comment]
> holy fuck the issue is bigger than I thought. if you even _view_ a stream that is at a non-standard framerate, it crashes everyone else's discord
> 
> basically, what I mean is, the client throws that error if you send _or_ receive a stream of an invalid framerate. If you remove the clientsided check, you can still stream at 144FPS, but it will crash everyone else's client if they view it.

I also fixed it myself and I tried lmao. Yep it was crashing for everyone else who tries to enter the stream.

**Comment by BlueberryWolf** [comment]
I managed to bypass it and avoid the crashes. I'll make a pull request.

**Comment by BlueberryWolf** [comment]
well actually no, that doesn't work. I tried spoofing MaxFrameRate, and keeping EncodingVideoFrameRate at what you set it to, but it turns out that MaxFrameRate is the max speed at which the client can *play back* the video stream. I have to rethink this.
